doc/fundamentals/spouts.md

```diff
@@ -0,0 +1,291 @@
+# Spouts
+
+## Introduction
+
+Spouts are a way to get streaming data from any source into Pachyderm.
+To create a spout, you need three things
+
+1. A source of streaming data, such as Kafka, nifi, rabbitMQ, etc.
+2. A containerized server.
```

Comment:
I don't know that I'd really call this a "server" -- this is a process that consumes data from some other streaming server source (so, if anything, it'd be a "client" of the streaming source, though I don't really think that term is particularly helpful in describing what's going on). I think at some point we'd like spouts to be able to be servers to, for example, receive webhook requests based on github events, but AFAIK we actually don't support this right now (is that right @adelelopez ?)

Reply:
Yeah, we don't support that yet. It is something I'd like to do though.

Reply:
Can I call it a daemon? :-) 

What should I call it?

Reply:
By the way, by my (admittedly old-guy) standards, a "server* is kind of like a *daemon*; it runs continously, disconnected from any tty, regardless of whether it actually serves data.  In this case, it's a piece of middleware, but I hate that term, too.